Build failure: python3Packages.tblite

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `nix-build -A python3Packages.tblite`

### Build log

```
....
Warning: Unused variable ‘b’ declared at (1) [-Wunused-variable]
../src/tblite/mesh/lebedev.f90:322:16:

  322 |    real(wp) :: a,b,v
      |                1
Warning: Unused variable ‘a’ declared at (1) [-Wunused-variable]
../src/tblite/mesh/lebedev.f90:322:18:

  322 |    real(wp) :: a,b,v
      |                  1
Warning: Unused variable ‘b’ declared at (1) [-Wunused-variable]
ninja: build stopped: subcommand failed.
error: builder for '/nix/store/clqsg83jjg292znb0g5kd3bc4rp2rjll-python3.11-tblite-0.3.0.drv' failed with exit code 1;
       last 10 log lines:
       >
       >   322 |    real(wp) :: a,b,v
       >       |                1
       > Warning: Unused variable ‘a’ declared at (1) [-Wunused-variable]
       > ../src/tblite/mesh/lebedev.f90:322:18:
       >
       >   322 |    real(wp) :: a,b,v
       >       |                  1
       > Warning: Unused variable ‘b’ declared at (1) [-Wunused-variable]
       > ninja: build stopped: subcommand failed.

```

### Additional context

Fails on latest master. 

### Notify maintainers

@sheepforce 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.59, NixOS, 23.05 (Stoat), 23.05.4448.5550a85a087c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.6`
```

--- a/pkgs/development/libraries/science/chemistry/tblite/python.nix
+++ b/pkgs/development/libraries/science/chemistry/tblite/python.nix
@@ -1,4 +1,5 @@
 { buildPythonPackage
+, fetchpatch
 , meson
 , ninja
 , pkg-config
@@ -42,8 +43,17 @@ buildPythonPackage {
 
   propagatedBuildInputs = [ tblite simple-dftd3 cffi numpy ];
 
-  # Add multicharge to the meson deps; otherwise we get missing mod_multicharge errors
-  patches = [ ./0001-fix-multicharge-dep-needed-for-static-compilation.patch ];
+
+  patches = [
+    # Add multicharge to the meson deps; otherwise we get missing mod_multicharge errors
+    ./0001-fix-multicharge-dep-needed-for-static-compilation.patch
+
+    # Toml-f 0.4.0 compatibility https://github.com/tblite/tblite/pull/108
+    (fetchpatch {
+      url = "https://github.com/tblite/tblite/commit/e4255519b58a5198a5fa8f3073bef1c78a4bbdbe.diff";
+      hash = "sha256-BMwYsdWfK+vG3BFgzusLYfwo0WXrYSPxJoEJIyOvbPg=";
+    })
+  ];
 
   format = "other";
   pythonImportsCheck = [ "tblite" "tblite.interface" ];

